src/app.rs

```diff
@@ -103,7 +97,19 @@ impl App {
             }
         };
 
-        let adapter = Adapter::new(session.clone()).await.unwrap();
+        // Start scanning if enabled
+        if config.station.auto_scan {
+            let iwd_station = session.station().unwrap();
+            match iwd_station.scan().await {
+                Ok(_) => (),
+                Err(e) => {
+                    error!("Failed to start scan");
+                    error!("{}", e.to_string());
+                }
+            }
+        }
```

Comment:
this will not work if the device mode is set to `access point` before starting impala.
You need to check for the device mode if you want keep this